### ISSUE TITLE
Feature Request: Add Fan curve

### ISSUE THREAD

**Issue by chetan1836** [Karmel0x/AsusFanControl#8]
The application works brilliantly, can you add a option for fan curve profile. Thanks.

**Comment by Karmel0x** [comment]
I'll do it someday, I'm too lazy for now. Especially since it won't be useful to myself tbh.

**Comment by kodifan** [comment]
+1 for that
or maybe at the start not fully working fan curve but a configurable parameter that will set a fan according of temp readings (that is how i handle it  now - i created  a task scheduler that is running every minute and it set fan to temperature reading minus 20 by calling dos executable).  Seem to be quite effective but if it would be done natively in the app (plus minimize to tray) it would be better :) and probably not require as much work as fan curve

**Comment by Leonm99** [comment]
I just stumbled across this bevause my asus tuf a15 has no other way for manual fan controll, im currently at work and got some spare time, im installing visual studio as i write this hehe. 

i will try and add a simple FanCurve editor, since i could use it as well for my laptop ;)

i will make a pull request once im done :)

**Comment by kodifan** [comment]
Somebody already did that
https://github.com/Darren80/AsusFanControlEnhanced
But you can always make it better;)

**Comment by Leonm99** [comment]
oh okay didnt see that, its sad to see he copied the repo and removed the original autor instead of just contributing to this project and making a pull request.. ill see what i can improve and make a pull request anyways, would be a dick move to just copy someones work without proper credit.

**Comment by kodifan** [comment]
probably you right but this repo is kind of dead for 4 months
for me its better to have an app with more features.  
i have even created  a  feature request on Darren80 fork :
https://github.com/Darren80/AsusFanControlEnhanced/issues/7
so you can also make such improvements, i will happily switch to your app :)

**Comment by Leonm99** [comment]
ah well. its not the end of the world its just a bit dissapointing, but yeah if its completely dead i can understand it for sure :) ill see what i can improve and make a pull request.

**Comment by kodifan** [comment]
hello @Leonm99  I see that you have committed two changes in Darens fork ? (one  "basic power/battery profiles" that I really wait for)
but he haven't approved this yet ?
maybe you will publish your release anyway ?
im afraid we can wait a long time (Daren haven't committed anything in his repo since late May)

**Comment by Leonm99** [comment]
@kodifan hello! I'm currently at work but I can compile the exe from my latest commit anyways. It will take a couple of hours till I can get to my PC and do it but expect it to be ready by tomorrow ;)

**Comment by kodifan** [comment]
@Leonm99  sure no hurry :) i will happily test it. please expect some other feature requests i have some ideas :)

**Comment by Leonm99** [comment]
> @Leonm99  sure no hurry :) i will happily test it. please expect some other feature requests i have some ideas :)

@kodifan yeah Sounds good to me. I'll try and implement stuff when i got time :)

**Comment by kodifan** [comment]
> > @Leonm99  sure no hurry :) i will happily test it. please expect some other feature requests i have some ideas :)
> 
> @kodifan yeah Sounds good to me. I'll try and implement stuff when i got time :)

its working fine if you ask (just first impression but its fine) 😄

**Comment by adamfluty** [comment]
Hi all! If you're still waiting for this fan control strategy to be implemented in a better tool, then try this special release of the GHelper app:
https://github.com/seerge/g-helper/discussions/2272

The power modes and fan curve controls work flawlessly on my Zenbook Pro Duo!!

**Comment by kodifan** [comment]
> Hi all! If you're still waiting for this fan control strategy to be implemented in a better tool (with curves and additional features), then try this special release of the GHelper app: [seerge/g-helper#2272](https://github.com/seerge/g-helper/discussions/2272)
> 
> The power modes and fan curve controls work flawlessly on my Zenbook Pro Duo!!

last time i checked (4 months ago) it didnt work at all for my asus but i will give yet another try...

**Comment by adamfluty** [comment]
That special release is from march. So make sure you use it instead of the official release. It uses the same drivers to drive the fan as the Nyasia app, so it should work in the same way the product here works.

**Comment by Leonm99** [comment]
thank you for showing that to us. i tested it and it works really nice. as much fun as it was to play around with the AsusFanControllEnhanced i think ill use this experimental release :) now i can get rid of armoury crate all together good stuff

**Comment by adamfluty** [comment]
Unlucky. I’m surprised one worked and not the other since they are the same drivers/tech. The only troubleshooting help i can think off is the recommendation they made to press the calibrate button. It ran my fans at a few levels then adjusted the axes of my curve graphs to match my machine. But nonetheless it’s good that this tool helped ya!

**Comment by pgain88** [comment]
> probably you right but this repo is kind of dead for 4 months for me its better to have an app with more features. i have even created a feature request on Darren80 fork : [Darren80/AsusFanControlEnhanced#7](https://github.com/Darren80/AsusFanControlEnhanced/issues/7) so you can also make such improvements, i will happily switch to your app :)

https://github.com/pgain88/AsusFanControlRemastered/releases/tag/V8

**Comment by left-winger502** [comment]
> https://github.com/pgain88/AsusFanControlRemastered

Seems kinda cool but just won't start on my pc for some reason

**Comment by Karmel0x** [comment]
> Seems kinda cool but just won't start on my pc for some reason

Probably you need to copy `run.bat` and `PsExec.exe` from [releases](https://github.com/Karmel0x/AsusFanControl/releases) and use it to run that fork.

**Comment by left-winger502** [comment]
@Karmel0x Yea, I've actually tried this and it indeed were able to launch just to crash a bit later so I've thought it wasn't the case lol

**Comment by realMoai** [comment]
Hello, i have made my own fan control utility that has a much more reliable curve system and more options to adjust how the fans react to it.
https://github.com/realMoai/ZenithFanUtility

**Comment by atharvshende2008** [comment]
> > probably you right but this repo is kind of dead for 4 months for me its better to have an app with more features. i have even created a feature request on Darren80 fork : [Darren80/AsusFanControlEnhanced#7](https://github.com/Darren80/AsusFanControlEnhanced/issues/7) so you can also make such improvements, i will happily switch to your app :)
> 
> https://github.com/pgain88/AsusFanControlRemastered/releases/tag/V8

bro i am unable to run the application
can you tell me step by step to run it .
